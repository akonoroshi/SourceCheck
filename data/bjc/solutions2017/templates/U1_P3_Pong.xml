<project name="U1_P3_Pong" app="iSnap 2.1, http://go.ncsu.edu/isnap" version="1"><data>{}</data><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAAHE0lEQVR4Xu2df0iVVxjHn3uvI02nrbJcfyRqahFBUCPIXOBSyJHOdAqbqw03bDGQHHMIcz8IYd4xaixINthklOBAlCBo9IMK2ZLAWtTKaixkEGZotWyLUY5zIOmijXPufd8ddv1cEJH7Pc/zno8f3veecy/vDUxMTEwIDwg4IhBAQEfkaasJICAiOCUwrYBnzpyR48eP+3ZgOTk5UlZW5lt9Cv9/CEwRsKurS1atWmU1g0UL06XntWxJCAb+dVxiQlAK2y/J6OiozqWmpsqCBQusehGOLwIRAp46dUpLodYlgUDA+PdAc6G8tDTZiExK25BcuHBB1x8ZGZH169cbjSMUnwQiBGxpaZHNmzdrOe7duydVVVXS3d0tW7dulf3790t5ebm0trZKb2+vnDhxQl+mlayXdxbLi7mzJwklvdUpf379yrTElIBnz56dlHvlypXxSZZZGRGIEHDXrl1SUFAwKcfw8LCon7S0NLlx44asXr1axsfHJTs7W+bNmycpKSly+vRp+e3T0ggBZ5XvlMDT6fLXvm1TDkIJ2N/fP3mGXbNmjdGBEopPAhECqtdmjy6PNpfh0S8r5YWsRCNCSsC+vj4tufrd3NxsNI5QfBKYsgjZu3ev5OXlWc1WnRGzsrKMxhw9elTnBgcHZfv27UZjCMUvgWm3Yc6fPy+HDx+Om1mrlxCbNm2S9PT0uJlTvExkioA9PT2yYsWK/2R+CzMWyfMvfyXBYMi3fge/3SJ374zo+gcOHJDGxkbfelHYnkCEgMeOHZOMjAzj7Reb14nTbeu82viDLMotsT9qyxHhbTK56Fm+fLnlaOJ+EnjiNox6N2Tjxo16lavetRgYGJCkpCR9LGolHAwG5ebNmzJ//nzZvXu33qqpq6vT2zUlJSVy8uRJuXr1qtTU1Ig6q+7YsUM6Ozulo6NDqqurZe7cufLmB32SkV0k34eXSSgkcufuA3njo8vyzcd5su/gsLxeliFbWgal+/Nl+rlQMKD//m5nvsxODErVexfli6YcaQj/GlHjmdQE+fHnOxLu+F0fb2vd32z7+GlRDLUjBFQirV27Vv+zzp07J6WlpRIOh/Vla86cOXL9+nXd6vbt29LQ0KAvadeuXZNbt27pvT0lqtobrK2tlbGxMS3qI0nVOPVcZWWlHDp0SPLz8+XtT/rlp953pKLxF+lqW6rPUtVNF3WPoufSpL7q2Wk3xAMBkT/GH0jirKAkhALyVEIgoob+eM/EhNS8f0nX+rD2Lts+MUji59AIAZUw6kynRCgsLIzoq85oj19G1SKluLhYy6pkUqtam3dP1Lh3P7so6ZkFfs5P126uGZs8A65bt873fjQwJzBlEaLOgmoRsmHDhogqR44cMa9qmMzNzZXFixcbpqOLDQ0NyZUrV+T+/fuyZMkS6y2m6LoyypTAEz+Opc5mjz/43KopUnI2BBDQhhZZzwkgoOdIKWhDAAFtaJH1nAACeo6UgjYEENCGFlnPCSCg50gpaEMAAW1okfWcAAJ6jpSCNgQQ0IYWWc8JIKDnSCloQwABbWiR9ZwAAnqOlII2BBDQhhZZzwkgoOdIKWhDAAFtaJH1nAACeo6UgjYEuD+gDS2ynhNAQM+RUtCGAALa0CLrOQEE9BwpBW0IIKANLbKeE0BAz5FS0IYAAtrQIus5AQT0HCkFbQggoA0tsp4TQEDPkVLQhgAC2tAi6zkBBPQcKQVtCCCgDS2ynhNAQM+RUtCGAALa0CKrCVRUVOgbku7Zs0eKiopiooKAMeGbeYOTk5P117g9eqhbNGdmZkYNAgGjRjczB6qb0z9+s9L29napr6+PGgYCRo1uZg4MhULy8OHDycmrb1FQ3yEY7QMBoyU3Q8epS676wnElYVNTk7S1tcVEAgFjwsfgWAkgYKwEGT8tAdOb3CMgAvlCAAF9wUpRUwIIaEqKnC8EENAXrBQ1JYCApqTI+UIAAX3BSlFTAghoSoqcLwQQ0BesFDUlgICmpMj5QgABfcFKUVMCCGhKipwvBBDQF6wUNSWAgKakyPlCAAF9wUpRUwIIaEqKnC8EENAXrBQ1JYCApqTI+UIAAX3BSlFTAghoSoqcUwJ8JN8pfpojIA44JYCATvHTHAFxwCkBBHSKn+YIiANOCSCgU/w0R0AccEoAAZ3ipzkC4oBTAgjoFD/NERAHnBJAQKf4aY6AOOCUAAI6xU9zBMQBpwQQ0Cl+miMgDjglgIBO8dMcAXHAKQEEdIqf5giIA04JIKBT/DRHQBxwSgABneKnOQLigFMCCOgUP80REAecEkBAp/hpjoA44JQAAjrFT3MExAGnBBDQKX6aIyAOOCWAgE7x0xwBccApAQR0ip/mCIgDTgkgoFP8NEdAHHBKAAGd4qc5AuKAUwII6BQ/zREQB5wSQECn+GmOgDjglMA/khcn1R4s/dwAAAAASUVORK5CYII=</thumbnail><stage name="Stage" guid="b38549a4-e050-4677-8887-101366bf7991" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="round" codify="false" inheritance="false" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAOhUlEQVR4Xu3VwQkAAAjEMN1/abewn7jAQRC64wgQIECAAIF3gX1fNEiAAAECBAiMAHsCAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQIHLFxAWmhEwHPAAAAAElFTkSuQmCC</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><sprite name="Ball" idx="2" x="186.60562578052804" y="28.168603284957783" heading="67.99999999999999" scale="0.4" rotation="1" draggable="true" costume="1" color="161.262,237.15,0" pen="tip" id="8"><costumes><list id="9"><item><costume name="Untitled" center-x="18" center-y="18" image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACQAAAAkCAYAAADhAJiYAAACUUlEQVRYR83YS6gOcRjH8c/ZWSlFrgt2lBTZOBQpSexcc8utWCBLJ5cFTuyIDTkot1x3pChRLhtRUuxYEKKUlR09mjlN0/vOzHvOO+95n93M/J/f851n/v//PP+nx9BsCZZiEsZjQiLzDd/xBQ/xqFX5nhYcerEXyzC6ot9vPMBpvKjiUwVoOvqwuYpgwZjLOI4PRTplQNtwYZggefftuNhMswjoGA60GSaV68fBRtrNgAYQb5LaK8wdJlxeIzK/I6/ZCCiyEtnphEWWIluDlgdaj2udIMnE2IDr6XUWKJbyJ4zpMNAvTEVsEbJAsST3JzDvMLNmsGyME8nWMggUhB9rBiiTnxZfKM3QIRwp86j5+WEcTYFiW59Xc8Ay+ZfoDaDJ+Fw2ukPPpwTQatzqUMCyMGsCaDfOlI3s0PM9AVTnP6vV9+gPoPON/imtKrVp/EAAncS+NgkOV+ZUAMXuHLt0N1hfAG3BpW6gwdYAWognXQK0KN2pf2DsCEP9xLgUKArwTSMMdCUOEinQWtxIgP7mypI6ObOx1uFmth56hvl1Ri/Qfo4F+QJtOe6NENAK3M8DxfUdrOww1F2sSmPmi/xRiPTNSQbEzG/36stqvk6myZ9mQHF/RgJVd7EfxX3M2ffZL9LsoLgYVzGxps/3FRvxOK9fdJSOojugouvRTotyOWAaHirKmg0BchY720R0DruKtKoAhf/s5NwU5e5Q7HZSUbwpc64KlOrMShpW0bSKn3KRPU2aVdGwelsGUrTKqvrG0TsmfaOWXkza/0fjVu0ft3taxhKU2YsAAAAASUVORK5CYII=" id="10"/></item></list></costumes><sounds><list id="11"></list></sounds><variables></variables><blocks></blocks><scripts><script x="362" y="87"><block id="260" s="receiveGo"></block><block id="261" s="setScale"><l>40</l></block><block id="278" s="gotoXY"><l>0</l><l>0</l></block><block id="441" s="setHeading"><block id="442" s="reportRandom"><l>45</l><l>135</l></block></block><block id="391" s="doForever"><script><block id="509" s="doIf"><block id="516" s="reportAnd"><block id="510" s="reportLessThan"><l>215</l><block id="511" s="xPosition"></block></block><block id="517" s="reportTouchingObject"><l><option>edge</option></l></block></block><script><block id="512" s="doChangeVar"><l>score</l><l>-1</l></block></script></block><block id="443" s="bounceOffEdge"></block><block id="440" s="forward"><l>10</l></block></script></block></script><script x="104" y="178"><block id="444" s="receiveCondition"><block id="463" s="reportTouchingObject"><l>Paddle 1</l></block></block><block id="467" s="setHeading"><block id="469" s="reportDifference"><l>360</l><block id="468" s="direction"></block></block></block><block id="507" s="doChangeVar"><l>score</l><l>1</l></block></script><script x="73" y="361"><block id="556" s="receiveCondition"><block id="557" s="reportTouchingObject"><l>Paddle 2</l></block></block><block id="558" s="setHeading"><block id="559" s="reportDifference"><l>360</l><block id="560" s="direction"></block></block></block></script></scripts></sprite><sprite name="Paddle 1" idx="1" x="215" y="-10" heading="90" scale="0.3" rotation="1" draggable="true" costume="2" color="80,80,80" pen="tip" id="71"><costumes><list id="72"><item><costume name="Untitled" center-x="18" center-y="83" image="data:," id="73"/></item><item><costume name="Paddle" center-x="20" center-y="96" image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAADACAYAAAB26SGCAAABWElEQVR4Xu3dQQqCYBRG0efO2lkuzaWF0UBoeIN+4TT/4HW8jt1m5jkzj1nzd2yfA/c175v9dgeekv/WvN5wP8HVUiRYnwhBglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBer+q8Fjge/IXm/wkvzkEa/6aeD3n3sBcNIi0aXV8YwAAAAASUVORK5CYII=" id="74"/></item></list></costumes><sounds><list id="75"></list></sounds><variables></variables><blocks></blocks><scripts><script x="143" y="78"><block id="58" s="receiveGo"></block><block id="84" s="setScale"><l>30</l></block><block id="102" s="gotoXY"><l>215</l><l>0</l></block><block id="506" s="doSetVar"><l>score</l><l>0</l></block><block id="168" s="doForever"><script><block id="169" s="doIf"><block id="279" s="reportAnd"><block id="188" s="reportKeyPressed"><l><option>up arrow</option></l></block><block id="280" s="reportLessThan"><block id="297" s="yPosition"></block><l>170</l></block></block><script><block id="189" s="changeYPosition"><l>5</l></block></script></block><block id="190" s="doIf"><block id="300" s="reportAnd"><block id="191" s="reportKeyPressed"><l><option>down arrow</option></l></block><block id="302" s="reportLessThan"><l>-170</l><block id="303" s="yPosition"></block></block></block><script><block id="192" s="changeYPosition"><l>-5</l></block></script></block></script></block></script><script x="237" y="420"><block id="167" s="changeYPosition"><l>10</l></block></script><script x="63" y="454"><block id="165" s="receiveKey"><l><option>up arrow</option></l></block></script></scripts></sprite><watcher var="score" style="normal" x="10" y="10" color="243,118,29"/><watcher scope="Ball" s="yPosition" style="normal" x="10" y="31.000001999999995" color="74,108,212"/><sprite name="Paddle 2" idx="3" x="-215" y="105" heading="90" scale="0.3" rotation="1" draggable="true" costume="2" color="80,80,80" pen="tip" id="127"><costumes><list id="128"><item><costume name="Untitled" center-x="18" center-y="83" image="data:," id="129"/></item><item><costume name="Paddle" center-x="20" center-y="96" image="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACgAAADACAYAAAB26SGCAAABWElEQVR4Xu3dQQqCYBRG0efO2lkuzaWF0UBoeIN+4TT/4HW8jt1m5jkzj1nzd2yfA/c175v9dgeekv/WvN5wP8HVUiRYnwhBglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBepegwSrQN1rkGAVqHsNEqwCda9BglWg7jVIsArUvQYJVoG61yDBKlD3GiRYBer+q8Fjge/IXm/wkvzkEa/6aeD3n3sBcNIi0aXV8YwAAAAASUVORK5CYII=" id="130"/></item></list></costumes><sounds><list id="131"></list></sounds><variables></variables><blocks></blocks><scripts><script x="143" y="78"><block id="519" s="receiveGo"></block><block id="520" s="setScale"><l>30</l></block><block id="521" s="gotoXY"><l>-215</l><l>0</l></block><block id="522" s="doSetVar"><l>score</l><l>0</l></block><block id="523" s="doForever"><script><block id="524" s="doIf"><block id="525" s="reportAnd"><block id="526" s="reportKeyPressed"><l><option>w</option></l></block><block id="527" s="reportLessThan"><block id="528" s="yPosition"></block><l>170</l></block></block><script><block id="529" s="changeYPosition"><l>5</l></block></script></block><block id="530" s="doIf"><block id="531" s="reportAnd"><block id="532" s="reportKeyPressed"><l><option>s</option></l></block><block id="533" s="reportLessThan"><l>-170</l><block id="534" s="yPosition"></block></block></block><script><block id="535" s="changeYPosition"><l>-5</l></block></script></block></script></block></script></scripts></sprite></sprites></stage><hidden></hidden><headers></headers><code></code><blocks></blocks><variables><variable name="score"><l>6</l></variable></variables><editing></editing></project>