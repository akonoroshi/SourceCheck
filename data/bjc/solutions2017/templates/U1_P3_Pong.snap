snapshot {
    @defVar (STAGE, START, SCORE, PADDLES) {
        @defVar (FOREVER) {
            stage {
                @defBlock PADDLE {
                    sprite {
                        script {
                            receiveGo
                            @anyOrder {
                                @optional { setScale(literal) }
                                gotoXY(literal, literal)
                            }
                            @if (FOREVER) {
                                doForever(script {
                                    doIf(reportAnd(reportKeyPressed(literal), reportLessThan(yPosition, literal)), script {
                                        changeYPosition(literal)
                                    })
                                    doIf(reportAnd(reportKeyPressed(literal), reportLessThan(literal, yPosition)), script {
                                        changeYPosition(literal)
                                    })
                                })
                            }
                        }
                        @unless (FOREVER) {
                            script {
                                receiveKey
                                doIf(reportLessThan(yPosition, literal), script {
                                    changeYPosition(literal)
                                })
                            }
                            script {
                                receiveKey
                                doIf(reportLessThan(literal, yPosition), script {
                                    changeYPosition(literal)
                                })
                            }
                        }
                    }
                }
                @block PADDLE
                sprite {
                    script {
                        receiveGo
                        @anyOrder {
                            @unless (STAGE, START) { doSetVar(literal, literal) }
                            @optional { setScale(literal) }
                            gotoXY(literal, literal)
                            setHeading(@anything)
                        }
                        doForever(script {
                            @unless (STAGE, START) {
                                doIf(reportAnd(reportLessThan(literal, xPosition), reportTouchingObject(literal)), script {
                                    doChangeVar(literal, literal)
                                })
                            }
                            bounceOffEdge
                            forward(literal)
                        })
                    }
                    @defVar DIFF {
                        script {
                            receiveCondition(reportTouchingObject(literal))
                            setHeading(
                                @if (DIFF) { reportProduct(direction, literal) }
                                @unless (DIFF) { reportDifference(literal, direction) }
                            )
                            @unless (STAGE, START) { doChangeVar(literal, literal) }
                        }
                        @if (STAGE, PADDLES) {
                            script {
                                receiveCondition(reportTouchingObject(literal))
                                setHeading(
                                    @if (DIFF) { reportProduct(direction, literal) }
                                    @unless (DIFF) { reportDifference(literal, direction) }
                                )
                            }
                        }
                    }
                }
                @if (STAGE, PADDLES) {
                    @block PADDLE
                }
            }
            @unless (STAGE, START) { var }
        }
    }
}